landoapi/hgexports.py

```diff
@@ -274,85 +287,83 @@ def __init__(self, fileobj: io.BytesIO):
         super().__init__(fileobj)
         self.parse_patch()
 
-    def parse_patch(self):
-        """Parse the Git patch file for headers and diff content."""
-        mail_from_line = next(self.patch)
-        if not mail_from_line.startswith(b"From "):
-            raise ValueError("Patch is malformed, first line is not `From `.")
-
-        author_from_line = next(self.patch)
-        if not author_from_line.startswith(b"From: "):
-            raise ValueError("Patch is malformed, second line is not `From:`.")
-        self.set_header(
-            "From", author_from_line.removeprefix(b"From: ").removesuffix(b"\n")
-        )
+    def get_header(self, name: bytes | str) -> Optional[str]:
+        """Get the headers from the message."""
+        name = name.decode("utf-8") if isinstance(name, bytes) else name
```

Comment:
Slightly less redundant:
```suggestion
        if isinstance(name, bytes):
            name = name.decode("utf-8")
```

Reply:
This is just carried over from the existing `PatchHelper` implementation. I'll update it.